src/modules/movie/movie.service.ts

```diff
@@ -13,11 +15,56 @@ export class MovieService {
   ) {}
 
   async getMovie(movieId: string) {
-    return await this.movieModel.findById(movieId);
+    const movie = await this.movieModel
+      .findById(movieId)
+      .populate('projections');
+
+    return movie;
   }
 
-  async getMovies() {
-    return await this.movieModel.find();
+  async getMovies(
+    cinemaId: string,
+    projections?: string,
+    date?: string,
+    projectionType?: ProjectionType,
+  ) {
+    const dateToPass = new Date(date);
+    const moviesWithProjections = await this.movieModel
+      .find({ cinemaId })
+      .populate(projections == 'true' ? 'projections' : null);
+    this.filterOutMovieProjections(
+      moviesWithProjections,
+      dateToPass,
+      projectionType,
+    );
+
+    return moviesWithProjections;
+  }
+
+  filterOutMovieProjections(
+    moviesWithProjections: Movie[],
+    date?: Date,
+    projectionType?: ProjectionType,
+  ) {
+    moviesWithProjections.forEach((movie) => {
+      movie.projections = movie.projections.filter((projection) => {
```

Comment:
Let's not modify the original object inside the methbod (that's a bad practice, code becomes untransperant and unpredictable). Better return a new object (you can create a shallow copy of the moviesWithProjections)